poly/joint.go

```diff
@@ -0,0 +1,284 @@
+package poly
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"io"
+)
+
+// This package provides  a dealer-less distributed verifiable secret sharing
+// using Pedersen VSS scheme as explained in "Provably Secure Distributed
+// Schnorr Signatures and a (t, n) Threshold Scheme for Implicit Certificates"
+// This file is only responsible for the setup of a shared secret among n peers.
+// The output is a global public polynomial (pubPoly) and a secret share for
+// each peers.
+
+// PolyInfo describe the information needed to construct (and verify) a matrixShare
+type PolyInfo struct {
+	// How many peer do we need to reconstruct a secret
+	T int
+	// How many peers do we need to verify
+	R int
+	// How many peers are collaborating into constructing the shared secret ( i.e. MatrixShare is of size NxN)
+	N int
+}
+
+// Represent the output of a VSS Pedersen scheme : a global public polynomial and a share of its related priv poly
+// for a peer
+type SharedSecret struct {
```

Comment:
On one hand, I think they should be public since the owner of this SharedSecret has the "right" to know its share, since it is its own. This struct is only a collection / holder of different pieces of information to represent the share of a node, and, in contrast, is not a fully oop-class-like struct with its own methods set and stuff, where private fields make sense. At least to me ;)
On the other hand, it is necessary that the pub poly, share & index stay constant and are not modified in any way by the holder of the struct otherwise the share would become invalid. In that way, it is preferable that they become private fields.
One trade off for these two ideas would be to make the fields private and make some Getter on it. Would that be ok for you ?